consensus/src/state_computer.rs

```diff
@@ -35,6 +36,18 @@ impl ExecutionProxy {
             synchronizer,
         }
     }
+
+    fn transactions_from_block(block: &Block<Vec<SignedTransaction>>) -> Vec<Transaction> {
+        let mut transactions = block
+            .payload()
+            .unwrap_or(&vec![])
+            .iter()
+            .map(|txn| Transaction::UserTransaction(txn.clone()))
```

Comment:
I think you could avoid doing these clones if you move the payload out of `block` and define an explicit constructor for the `BlockMetadata` instead of implementing the `From` trait? 

In addition, `From` takes a `&Block`, which forces it to clone the validator signatures. The explicit constructor would let you move them.

Reply:
we intentionally made block immutable, as unfortunate result we need to clone the transaction. If the clone becomes a bottleneck, we'll probably replace it with an Arc